frontend/packages/ux-editor/src/components/config/editModal/EditImage/ExternalImage/ExternalImage.tsx

```diff
@@ -88,5 +65,46 @@ export const ExternalImage = ({
   );
 };
 
+type EditUrlProps = {
+  url: string;
+  existingImageUrl: string;
+  onBlur: (event: React.ChangeEvent<HTMLInputElement>) => void;
+};
+
+const EditUrl = ({ url, existingImageUrl, onBlur }: EditUrlProps): React.ReactElement => {
+  const { t } = useTranslation();
+  const [isViewMode, setIsViewMode] = useState<boolean>();
+  const label = t('ux_editor.properties_panel.images.enter_external_url');
+  const noUrlText = t('ux_editor.properties_panel.images.external_url_not_added');
+  const value = calculateViewValue(url, noUrlText, isViewMode);
+
+  return isInitialUrlProvided(url, existingImageUrl) ? (
+    <StudioToggleableTextfield
+      onIsViewMode={setIsViewMode}
+      Icon={LinkIcon}
+      label={label}
+      onBlur={onBlur}
+      title={url}
+      value={value}
+    />
+  ) : (
+    <StudioIconTextfield label={label} value={url} onBlur={onBlur} Icon={LinkIcon} />
+  );
```

Comment:
Jeg har vanskeligheter med å forstå hva som skal skje her. Hvorfor laste to forskjellige komponenter avhengig av `isInitialUrlProvided`? Og hvorfor heter funksjonen `initial` når den kjøres hver gang komponenten blir lastet med nye verdier?

Reply:
Slik jeg tolker dette høres det ut som at vi bør gjøre komponenten kontrollerbar. Det vil si å styre tilstanden fra utsiden ved å eksponere `isViewMode` som en prop. Vi kan bruke den egendefinerte hooken `usePropState` til å gjøre at komponenten kan være både kontrollert og ukontrollert. Alternativt kan vi håndtere dette på innsiden av komponenten ved sende inn en `defaultMode`-prop eller noe slikt.

Men hvis dette uansett ikke er så viktig, har i hvert fall ikke jeg noe imot at vi fjerner denne kompleksiteten fullstendig.

Reply:
Kan sjekke med @Annikenkbrathen 😊